trino-rest-github/src/main/java/pl/net/was/rest/github/Sync.java

```diff
@@ -1852,12 +1855,22 @@ private static int getEmptyLimit()
         return Integer.parseInt(limit);
     }
 
+    private static long getQueryDelayMillis()
+    {
+        String queryDelay = System.getenv("QUERY_DELAY_MS");
+        if (queryDelay == null) {
+            return 0L;
+        }
+        return Long.parseLong(queryDelay);
+    }
+
     private static int retryExecute(PreparedStatement statement)
             throws SQLException
     {
         int breaker = 3;
         while (true) {
             try {
+                MILLISECONDS.sleep(getQueryDelayMillis());
```

Comment:
Sleep for (delay - query duration) after running the query, if the value is greater than zero